### PR TITLE
improve test failure msgs

### PR DESCRIPTION
**Pull request by aniketmaurya**
<details>
  <summary><b>Before submitting</b></summary>

- [ ] Was this discussed/agreed via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

</details>

## What does this PR do?

Fixes # (issue).

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in GitHub issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/test_lit_server.py [hunk 1]
```diff
@@ -17,14 +16,14 @@ def test_new_pipe(lit_server):
     for _ in range(pool_size):
         lit_server.new_pipe()
 
-    assert len(lit_server.pipe_pool) == 0
-    assert len(lit_server.new_pipe()) == 2
+    assert len(lit_server.pipe_pool) == 0, "pipe_pool was completely used and need to be empty"
```
Comment (williamFalcon) [inline tests/test_lit_server.py:19]: can you describe it in non technical terms as well?
- what in the code caused this issue?
- etc...

File: tests/test_lit_server.py [hunk 2]
```diff
@@ -101,7 +100,7 @@ def test_run():
     )
 
     time.sleep(5)
-    assert os.path.exists("client.py")
+    assert os.path.exists("client.py"), f"Expected client file to be created at {os.getcwd()} after starting the server"
```
Comment (williamFalcon) [inline tests/test_lit_server.py:103]: yes, this is a great comment

File: tests/test_lit_server.py [hunk 3]
```diff
@@ -101,7 +100,7 @@ def test_run():
...
     output = subprocess.run("python client.py", shell=True, capture_output=True, text=True).stdout
-    assert '{"output":16.0}' in output
+    assert '{"output":16.0}' in output, "tests/simple_server.py didn't return expected output"
```
Comment (williamFalcon) [inline tests/test_lit_server.py:105]: great comment

File: tests/test_lit_server.py (hunk 1 above)
Comment (aniketmaurya) [reply tests/test_lit_server.py]: @williamFalcon updated to

```py
assert len(lit_server.pipe_pool) == 0, "lit_server.new_pipe() was used till max_pool_size which exhausts the pool"
assert len(lit_server.new_pipe()) == 2, "lit_server.new_pipe() always must return a tuple of read and write pipes"
```

**Pull request merged**